### PR TITLE
Add tests to help keep tox and github CI configurations approximately in sync

### PR DESCRIPTION
**Pull request by bwhmather**

### CODE REVIEW

**Review by jgberry — APPROVED** [review]
Wow, now we have tests for our tests. This is getting very meta indeed :laughing: I think this does address some of my concerns with maintainability, although I question whether or not there is a better way to do this. I'm not aware of what that better approach would look like though, so at least for the time being, this will do.

**Comment by cgahr** [comment]
I think the issue is that there are 2 different ways to test the code, one using tox, the other using github actions. I think one of those should go and then we indirectly loose the duplication.

My proposal would be the following:
- all linters (black, isort, pylint, pyflakes, maybe mypy) go into pre-commit.
- we add a new github ci that runs pre-commit
- we add proper pytest configuration to test the code locally with the installed python version
- we run a matrix test for different OS and different python versions using github ci (as it is right now)

Basically this boils down to removing tox.

Alternatively, we could use https://github.com/ymyzk/tox-gh-actions to run tox using github actions.

**Comment by bwhmather** [comment]
Internet points will be awarded for tests that verify that these tests work.  Config file parsing probably a good candidate. :D

**Comment by bwhmather** [comment]
Merging this one.  Have raised an issue so that we can continue this discussion without comments disappearing as work is done: https://github.com/bwhmather/ssort/issues/87
